superset-frontend/src/components/Datasource/DatasourceModal.tsx

```diff
@@ -101,6 +101,10 @@ const DatasourceModal: FunctionComponent<DatasourceModalProps> = ({
       postPayload: {
         data: {
           ...currentDatasource,
+          cache_timeout:
```

Comment:
Can you also add this logic to the server-side endpoint to make sure this treatment is applied when invoking the API directly?

Reply:
Currently the Datasource modal is unfortunately using the legacy endpoint, which is why this check isn't happening in the API. If it were in fact using the V1 API, the old request would produce an error, as it does not satisfy the Marshmallow schema.